src/containers/booking-confirmation/BookingRewards.tsx

```diff
@@ -1,41 +1,47 @@
-import { Grid, Box, Typography, Alert } from '@mui/material';
+import { Alert, Box, Grid, Typography } from '@mui/material';
 import { useEffect, useState } from 'react';
 import { useRewards } from 'src/hooks/useRewards';
 import { RewardCard } from './RewardCard';
 import { RewardModal } from './RewardModal';
-import { useSearchParams } from 'react-router-dom';
+import { getOfferId } from 'src/hooks/useCheckout/helpers';
+import { useCheckout } from 'src/hooks/useCheckout';
 
 const convertTonsToKilos = (tons: string | undefined): number => {
   if (!tons) return 0;
-
-  const kilos = Number(tons) * 1000;
-  return kilos;
+  return Number(tons) * 1000;
 };
 
 const RewardIntroduction = ({ children }) => {
+  const { bookingMode } = useCheckout();
+  const isGroupMode = bookingMode === 'group';
   return (
     <Box mt={6} mb={6}>
       <Typography variant="h4" component="h3">
         Choose your reward option
       </Typography>
       <Typography variant="body1" mb={3}>
-        Numbers are estimated based on your booking value.
+        Numbers are estimated based on your booking value.{' '}
+        {isGroupMode && 'They might change after your offer confirmation'}
       </Typography>
       {children}
     </Box>
   );
 };
 
 export const BookingRewards = () => {
-  const [params] = useSearchParams();
-  const offerId = params.get('offerId');
-  const { data, isLoading, claimReward, error } = useRewards(offerId);
+  const { bookingMode, bookingInfo } = useCheckout();
+  const isGroupMode = bookingMode === 'group';
+  const offerId = !isGroupMode ? getOfferId(bookingInfo?.offers) : undefined;
+  const queryId = isGroupMode ? bookingInfo?.requestId : offerId;
```

Comment:
can be simplified, `queryId = isGroupMode ? bookingInfo?.requestId : getOfferId(bookingInfo?.offers);`